### ISSUE TITLE
Support PyCharm 2023.3 EAP

### ISSUE THREAD

**Issue by jerzyk** [trinhanhngoc/pycharm-odoo#415]
PyCharm 2023.3 EAP (Professional Edition)
Build #PY-233.6745.319, built on September 25, 2023
Runtime version: 17.0.8.1+7-b1059.3 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 13.5.2
GC: G1 Young Generation, G1 Old Generation

```
com.intellij.diagnostic.PluginException: Override `execute` (class=class dev.ngocta.pycharm.odoo.OdooApplicationInitializedListener) [Plugin: dev.ngocta.pycharm-odoo]
	at com.intellij.ide.ApplicationInitializedListener.execute$suspendImpl(ApplicationInitializedListener.kt:29)
	at com.intellij.ide.ApplicationInitializedListener.execute(ApplicationInitializedListener.kt)
	at com.intellij.ide.bootstrap.ApplicationLoader$callAppInitialized$1.invokeSuspend(ApplicationLoader.kt:450)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
```

```
java.lang.Throwable: Slow operations are prohibited on EDT. See SlowOperations.assertSlowOperationsAreAllowed javadoc.
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:376)
	at com.intellij.util.SlowOperations.assertSlowOperationsAreAllowed(SlowOperations.java:95)
	at com.intellij.workspaceModel.core.fileIndex.impl.WorkspaceFileIndexDataImpl.ensureIsUpToDate(WorkspaceFileIndexDataImpl.kt:128)
	at com.intellij.workspaceModel.core.fileIndex.impl.WorkspaceFileIndexDataImpl.getFileInfo(WorkspaceFileIndexDataImpl.kt:74)
	at com.intellij.workspaceModel.core.fileIndex.impl.WorkspaceFileIndexImpl.getFileInfo(WorkspaceFileIndexImpl.kt:239)
	at com.intellij.workspaceModel.core.fileIndex.impl.WorkspaceFileIndexImpl.findFileSetWithCustomData(WorkspaceFileIndexImpl.kt:224)
	at com.intellij.openapi.roots.impl.ProjectFileIndexImpl.getModuleForFile(ProjectFileIndexImpl.java:97)
	at com.intellij.openapi.roots.impl.ProjectFileIndexImpl.getModuleForFile(ProjectFileIndexImpl.java:90)
	at com.intellij.openapi.module.ModuleUtilCore.findModuleForPsiElement(ModuleUtilCore.java:151)
	at dev.ngocta.pycharm.odoo.OdooUtils.getProjectModuleWithDependenciesScope(OdooUtils.java:56)
	at dev.ngocta.pycharm.odoo.data.OdooExternalIdIndex.findRecordsById(OdooExternalIdIndex.java:241)
	at dev.ngocta.pycharm.odoo.xml.dom.OdooDomViewElement.getInheritedViewArch(OdooDomViewElement.java:161)
	at dev.ngocta.pycharm.odoo.xml.dom.OdooDomViewElement.lambda$locateInheritedElement$0(OdooDomViewElement.java:216)
	at dev.ngocta.pycharm.odoo.OdooUtils.getParameterizedCachedValue(OdooUtils.java:277)
	at dev.ngocta.pycharm.odoo.OdooUtils.getParameterizedCachedValue(OdooUtils.java:288)
	at dev.ngocta.pycharm.odoo.OdooUtils.getCachedValue(OdooUtils.java:261)
	at dev.ngocta.pycharm.odoo.xml.dom.OdooDomViewElement.locateInheritedElement(OdooDomViewElement.java:211)
	at dev.ngocta.pycharm.odoo.xml.dom.OdooDomViewInheritLocator.getInheritedDomElement(OdooDomViewInheritLocator.java:29)
	at dev.ngocta.pycharm.odoo.xml.dom.OdooDomViewInheritLocator.lambda$getInheritedViewModel$0(OdooDomViewInheritLocator.java:43)
	at dev.ngocta.pycharm.odoo.OdooUtils.getParameterizedCachedValue(OdooUtils.java:277)
	at dev.ngocta.pycharm.odoo.OdooUtils.getParameterizedCachedValue(OdooUtils.java:288)
	at dev.ngocta.pycharm.odoo.xml.dom.OdooDomViewInheritLocator.getInheritedViewModel(OdooDomViewInheritLocator.java:42)
	at dev.ngocta.pycharm.odoo.xml.dom.OdooDomModelScopedViewElement.lambda$getModel$0(OdooDomModelScopedViewElement.java:26)
	at dev.ngocta.pycharm.odoo.OdooUtils.getParameterizedCachedValue(OdooUtils.java:277)
	at dev.ngocta.pycharm.odoo.OdooUtils.getParameterizedCachedValue(OdooUtils.java:288)
	at dev.ngocta.pycharm.odoo.OdooUtils.getCachedValue(OdooUtils.java:261)
	at dev.ngocta.pycharm.odoo.xml.dom.OdooDomModelScopedViewElement.getModel(OdooDomModelScopedViewElement.java:16)
	at dev.ngocta.pycharm.odoo.python.OdooPythonLanguageInjector$3.accepts(OdooPythonLanguageInjector.java:195)
	at dev.ngocta.pycharm.odoo.python.OdooPythonLanguageInjector$3.accepts(OdooPythonLanguageInjector.java:131)
	at com.intellij.patterns.ObjectPattern.accepts(ObjectPattern.java:48)
	at com.intellij.patterns.ObjectPattern.accepts(ObjectPattern.java:37)
	at dev.ngocta.pycharm.odoo.python.OdooPythonLanguageInjector.getLanguagesToInject(OdooPythonLanguageInjector.java:211)
	at com.intellij.psi.impl.source.tree.injected.InjectedLanguageManagerImpl.processInPlaceInjectorsFor(InjectedLanguageManagerImpl.java:495)
	at com.intellij.psi.impl.source.tree.injected.InjectedLanguageUtilBase.probeElementsUpInner(InjectedLanguageUtilBase.java:237)
	at com.intellij.psi.impl.source.tree.injected.InjectedLanguageUtilBase.lambda$probeElementsUp$0(InjectedLanguageUtilBase.java:217)
	at com.intellij.openapi.application.impl.ReadActionCacheImpl$allowInWriteAction$1.invoke(ReadActionCacheImpl.kt:41)
	at com.intellij.openapi.application.impl.ReadActionCacheImpl$allowInWriteAction$1.invoke(ReadActionCacheImpl.kt:41)
	at com.intellij.openapi.application.impl.ReadActionCacheImpl.allowInWriteAction(ReadActionCacheImpl.kt:33)
	at com.intellij.openapi.application.impl.ReadActionCacheImpl.allowInWriteAction(ReadActionCacheImpl.kt:41)
	at com.intellij.psi.impl.source.tree.injected.InjectedLanguageUtilBase.probeElementsUp(InjectedLanguageUtilBase.java:216)
	at com.intellij.psi.impl.source.tree.injected.InjectedLanguageUtilBase.enumerate(InjectedLanguageUtilBase.java:158)
	at com.intellij.psi.impl.source.tree.injected.InjectedLanguageUtilBase.enumerate(InjectedLanguageUtilBase.java:128)
	at com.intellij.psi.impl.source.tree.injected.InjectedLanguageManagerImpl.enumerate(InjectedLanguageManagerImpl.java:371)
	at com.intellij.psi.impl.source.tree.injected.InjectedLanguageManagerImpl.getInjectedPsiFiles(InjectedLanguageManagerImpl.java:509)
	at com.intellij.psi.impl.smartPointers.InjectedSelfElementInfo.getInjectedFileIn(InjectedSelfElementInfo.java:130)
	at com.intellij.psi.impl.smartPointers.InjectedSelfElementInfo.restoreElement(InjectedSelfElementInfo.java:96)
	at com.intellij.psi.impl.smartPointers.SmartPsiElementPointerImpl.doRestoreElement(SmartPsiElementPointerImpl.java:86)
	at com.intellij.psi.impl.smartPointers.SmartPsiElementPointerImpl.getElement(SmartPsiElementPointerImpl.java:77)
	at com.intellij.codeInsight.folding.impl.EditorFoldingInfo.getPsiElement(EditorFoldingInfo.java:51)
	at com.intellij.codeInsight.folding.impl.UpdateFoldRegionsOperation.lambda$shouldRemoveRegion$1(UpdateFoldRegionsOperation.java:268)
	at com.intellij.util.SlowOperations.allowSlowOperations(SlowOperations.java:186)
	at com.intellij.codeInsight.folding.impl.UpdateFoldRegionsOperation.shouldRemoveRegion(UpdateFoldRegionsOperation.java:268)
	at com.intellij.codeInsight.folding.impl.UpdateFoldRegionsOperation.removeInvalidRegions(UpdateFoldRegionsOperation.java:207)
	at com.intellij.codeInsight.folding.impl.UpdateFoldRegionsOperation.run(UpdateFoldRegionsOperation.java:84)
	at com.intellij.openapi.editor.impl.FoldingModelImpl.runBatchFoldingOperation(FoldingModelImpl.java:292)
	at com.intellij.openapi.editor.impl.FoldingModelImpl.runBatchFoldingOperation(FoldingModelImpl.java:233)
	at com.intellij.openapi.editor.FoldingModel.runBatchFoldingOperationDoNotCollapseCaret(FoldingModel.java:107)
	at com.intellij.codeInsight.folding.impl.FoldingUpdate.lambda$getUpdateResult$1(FoldingUpdate.java:100)
	at com.intellij.codeInsight.folding.impl.CodeFoldingManagerImpl.lambda$updateFoldRegionsAsync$5(CodeFoldingManagerImpl.java:220)
	at com.intellij.codeInsight.folding.impl.CodeFoldingPass.doApplyInformationToEditor(CodeFoldingPass.java:44)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.applyInformationToEditor(TextEditorHighlightingPass.java:114)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService.lambda$applyInformationToEditorsLater$1(PassExecutorService.java:488)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:208)
	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:21)
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:190)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:844)
	at com.intellij.openapi.application.impl.ApplicationImpl$4.run(ApplicationImpl.java:464)
	at com.intellij.openapi.application.impl.RwLockHolder.runWithEnabledImplicitRead(RwLockHolder.kt:75)
	at com.intellij.openapi.application.impl.RwLockHolder.runWithImplicitRead(RwLockHolder.kt:67)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWithImplicitRead(ApplicationImpl.java:1431)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:82)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:124)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:44)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:792)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:739)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:733)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:761)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:677)
	at com.intellij.ide.IdeEventQueue._dispatchEvent$lambda$12(IdeEventQueue.kt:579)
	at com.intellij.openapi.application.impl.RwLockHolder.runWithoutImplicitRead(RwLockHolder.kt:44)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:579)
	at com.intellij.ide.IdeEventQueue.access$_dispatchEvent(IdeEventQueue.kt:70)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:350)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:349)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:793)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:349)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:344)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$1(IdeEventQueue.kt:989)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:105)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:989)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$7(IdeEventQueue.kt:344)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:849)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:386)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)

```

```
com.intellij.openapi.project.IndexNotReadyException: Please change caller according to com.intellij.openapi.project.IndexNotReadyException documentation
	at com.intellij.openapi.project.IndexNotReadyException.create(IndexNotReadyException.java:67)
	at com.intellij.util.indexing.FileBasedIndexImpl.handleDumbMode(FileBasedIndexImpl.java:815)
	at com.intellij.util.indexing.FileBasedIndexImpl.ensureUpToDate(FileBasedIndexImpl.java:763)
	at com.intellij.util.indexing.FileBasedIndexEx.processExceptions(FileBasedIndexEx.java:307)
	at com.intellij.util.indexing.FileBasedIndexEx.getContainingFilesIterator(FileBasedIndexEx.java:255)
	at com.intellij.util.indexing.FileBasedIndexEx.getContainingFiles(FileBasedIndexEx.java:231)
	at com.intellij.util.indexing.FileBasedIndexImpl.getContainingFiles(FileBasedIndexImpl.java:1067)
	at dev.ngocta.pycharm.odoo.module.OdooModuleIndex.getOdooModuleByName(OdooModuleIndex.java:57)
	at dev.ngocta.pycharm.odoo.module.OdooModuleIndex.getOdooModuleByName(OdooModuleIndex.java:70)
	at dev.ngocta.pycharm.odoo.module.OdooModuleUtils.getSystemWideOdooModules(OdooModuleUtils.java:174)
	at dev.ngocta.pycharm.odoo.module.OdooModule.lambda$getOdooModuleScope$7(OdooModule.java:222)
	at dev.ngocta.pycharm.odoo.OdooUtils.getParameterizedCachedValue(OdooUtils.java:277)
	at dev.ngocta.pycharm.odoo.OdooUtils.getParameterizedCachedValue(OdooUtils.java:288)
	at dev.ngocta.pycharm.odoo.module.OdooModule.getOdooModuleScope(OdooModule.java:217)
	at dev.ngocta.pycharm.odoo.module.OdooModule.getOdooModuleWithDependenciesScope(OdooModule.java:196)
	at dev.ngocta.pycharm.odoo.module.OdooModuleUtils.getOdooModuleWithDependenciesOrSystemWideModulesScope(OdooModuleUtils.java:207)
	at dev.ngocta.pycharm.odoo.model.OdooModelIndex.lambda$getOdooModelClassDefinitionsByName$2(OdooModelIndex.java:133)
	at dev.ngocta.pycharm.odoo.OdooUtils.getParameterizedCachedValue(OdooUtils.java:277)
	at dev.ngocta.pycharm.odoo.OdooUtils.getParameterizedCachedValueOnModule(OdooUtils.java:304)
	at dev.ngocta.pycharm.odoo.model.OdooModelIndex.getOdooModelClassDefinitionsByName(OdooModelIndex.java:132)
	at dev.ngocta.pycharm.odoo.model.OdooModelClass.getDefinitions(OdooModelClass.java:66)
	at dev.ngocta.pycharm.odoo.model.OdooModelClass.getSuperClasses(OdooModelClass.java:145)
	at dev.ngocta.pycharm.odoo.model.OdooModelClass.lambda$doGetExplicitAncestorClasses$2(OdooModelClass.java:113)
	at dev.ngocta.pycharm.odoo.OdooUtils.getParameterizedCachedValue(OdooUtils.java:277)
	at dev.ngocta.pycharm.odoo.OdooUtils.getParameterizedCachedValueOnModule(OdooUtils.java:304)
	at dev.ngocta.pycharm.odoo.model.OdooModelClass.doGetExplicitAncestorClasses(OdooModelClass.java:111)
	at dev.ngocta.pycharm.odoo.model.OdooModelClass.getExplicitAncestorClasses(OdooModelClass.java:99)
	at dev.ngocta.pycharm.odoo.model.OdooModelClass.lambda$getAncestorClasses$0(OdooModelClass.java:76)
	at dev.ngocta.pycharm.odoo.OdooUtils.getParameterizedCachedValue(OdooUtils.java:277)
	at dev.ngocta.pycharm.odoo.OdooUtils.getParameterizedCachedValueOnModule(OdooUtils.java:304)
	at dev.ngocta.pycharm.odoo.model.OdooModelClass.getAncestorClasses(OdooModelClass.java:75)
	at dev.ngocta.pycharm.odoo.model.OdooModelClassType.getAncestorTypes(OdooModelClassType.java:185)
	at com.jetbrains.python.codeInsight.stdlib.PyNamedTupleTypeProvider$Companion.getFieldTypeForTypingNTFunctionInheritor(PyNamedTupleTypeProvider.kt:97)
	at com.jetbrains.python.codeInsight.stdlib.PyNamedTupleTypeProvider$Companion.access$getFieldTypeForTypingNTFunctionInheritor(PyNamedTupleTypeProvider.kt:53)
	at com.jetbrains.python.codeInsight.stdlib.PyNamedTupleTypeProvider.getReferenceExpressionType(PyNamedTupleTypeProvider.kt:36)
	at com.jetbrains.python.psi.impl.PyReferenceExpressionImpl.getTypeFromProviders(PyReferenceExpressionImpl.java:379)
	at com.jetbrains.python.psi.impl.PyReferenceExpressionImpl.getType(PyReferenceExpressionImpl.java:215)
	at com.jetbrains.python.psi.types.TypeEvalContext.lambda$getType$0(TypeEvalContext.java:181)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:111)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:27)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:66)
	at com.jetbrains.python.psi.types.TypeEvalContext.getType(TypeEvalContext.java:170)
	at com.jetbrains.python.psi.impl.references.PyOperatorReference.resolveMember(PyOperatorReference.java:126)
	at com.jetbrains.python.psi.impl.references.PyOperatorReference.resolveLeftAndRightOperators(PyOperatorReference.java:104)
	at com.jetbrains.python.psi.impl.references.PyOperatorReference.resolveInner(PyOperatorReference.java:51)
	at com.jetbrains.python.psi.impl.references.PyReferenceImpl.multiResolveInner(PyReferenceImpl.java:107)
	at com.jetbrains.python.psi.impl.references.PyReferenceImpl$CachingResolver.resolve(PyReferenceImpl.java:775)
	at com.jetbrains.python.psi.impl.references.PyReferenceImpl$CachingResolver.resolve(PyReferenceImpl.java:770)
	at com.intellij.psi.impl.source.resolve.ResolveCache.lambda$resolveWithCaching$1(ResolveCache.java:159)
	at com.intellij.openapi.util.Computable.get(Computable.java:16)
	at com.intellij.psi.impl.source.resolve.ResolveCache.lambda$loggingResolver$4(ResolveCache.java:234)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:111)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:27)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:66)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolve(ResolveCache.java:212)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:158)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:145)
	at com.jetbrains.python.psi.impl.references.PyReferenceImpl.multiResolve(PyReferenceImpl.java:94)
	at com.jetbrains.python.psi.impl.PyCallExpressionHelper.multiResolveCallee(PyCallExpressionHelper.java:204)
	at com.jetbrains.python.psi.impl.PyCallExpressionHelper.getCallType(PyCallExpressionHelper.java:613)
	at com.jetbrains.python.psi.impl.PyBinaryExpressionImpl.getType(PyBinaryExpressionImpl.java:139)
	at com.jetbrains.python.psi.types.TypeEvalContext.lambda$getType$0(TypeEvalContext.java:181)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:111)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:27)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:66)
	at com.jetbrains.python.psi.types.TypeEvalContext.getType(TypeEvalContext.java:170)
	at com.jetbrains.python.codeInsight.typing.PyTypingTypeProvider.getClassType(PyTypingTypeProvider.java:1013)
	at com.jetbrains.python.codeInsight.typing.PyTypingTypeProvider.getTypeForResolvedElement(PyTypingTypeProvider.java:872)
	at com.jetbrains.python.codeInsight.typing.PyTypingTypeProvider.getType(PyTypingTypeProvider.java:707)
	at com.jetbrains.python.codeInsight.typing.PyTypingTypeProvider.lambda$getType$17(PyTypingTypeProvider.java:699)
	at com.jetbrains.python.codeInsight.typing.PyTypingTypeProvider.staticWithCustomContext(PyTypingTypeProvider.java:1882)
	at com.jetbrains.python.codeInsight.typing.PyTypingTypeProvider.getType(PyTypingTypeProvider.java:699)
	at com.jetbrains.python.validation.CompatibilityVisitor.checkBitwiseOrUnionSyntax(CompatibilityVisitor.java:767)
	at com.jetbrains.python.validation.CompatibilityVisitor.visitPyBinaryExpression(CompatibilityVisitor.java:211)
	at com.jetbrains.python.psi.impl.PyBinaryExpressionImpl.acceptPyVisitor(PyBinaryExpressionImpl.java:36)
	at com.jetbrains.python.psi.impl.PyBaseElementImpl.accept(PyBaseElementImpl.java:61)
	at com.jetbrains.python.validation.PyAnnotator.annotateElement(PyAnnotator.java:44)
	at com.jetbrains.python.validation.UnsupportedFeatures.annotateElement(UnsupportedFeatures.java:46)
	at com.jetbrains.python.validation.PyAnnotatingVisitor.runAnnotators(PyAnnotatingVisitor.java:61)
	at com.jetbrains.python.validation.PyAnnotatingVisitor.annotate(PyAnnotatingVisitor.java:54)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.runAnnotators(DefaultHighlightVisitor.java:128)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.visit(DefaultHighlightVisitor.java:106)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.runVisitors(GeneralHighlightingPass.java:361)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.lambda$collectHighlights$7(GeneralHighlightingPass.java:300)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.analyzeByVisitors(GeneralHighlightingPass.java:321)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.lambda$analyzeByVisitors$8(GeneralHighlightingPass.java:324)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.analyze(DefaultHighlightVisitor.java:91)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.analyzeByVisitors(GeneralHighlightingPass.java:324)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.collectHighlights(GeneralHighlightingPass.java:287)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.collectInformationWithProgress(GeneralHighlightingPass.java:235)
	at com.intellij.codeInsight.daemon.impl.ProgressableTextEditorHighlightingPass.doCollectInformation(ProgressableTextEditorHighlightingPass.java:95)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:57)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$1(PassExecutorService.java:406)
	at com.intellij.platform.diagnostic.telemetry.helpers.TraceKt.runWithSpanIgnoreThrows(trace.kt:76)
	at com.intellij.platform.diagnostic.telemetry.helpers.TraceUtil.runWithSpanThrows(TraceUtil.java:34)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:401)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1076)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$3(PassExecutorService.java:392)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:610)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:685)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:641)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:609)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:76)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:391)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:367)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:200)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:185)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:365)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:187)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)

```

**Comment by trinhanhngoc** [comment]
Hi @jerzyk , The plugin does not support PyCharm 2023.3 yet. I will add support for PyCharm 2023.3 in the next release.

**Comment by jerzyk** [comment]
thanks for the info, fot the future EAP release - shall I gather those exceptions for you, or just errors from main releases?

**Comment by trinhanhngoc** [comment]
> thanks for the info, fot the future EAP release - shall I gather those exceptions for you, or just errors from main releases?

Next week, I will release a new update for the Odoo plugin with support for PyCharm 2023.3 EAP. Then you can test and report any errors with EAP.

**Comment by trinhanhngoc** [comment]
Hi @jerzyk , The new version 2023.9.2 has been released with support for PyCharm 2023.3 EAP.

**Comment by jerzyk** [comment]
thsnks, updating & testing :)
